src/languages/java.md

```diff
@@ -1,3 +1,101 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Java is the most popular language in the globe, and it is in several fields:
+
+* Mobile applications
+* Desktop applications
+* Web applications
+* Web servers and application servers
+* Games
+* Database connection
+* Database
+* And much, much more!
+
+
+## Supported versions
+
+* 8
+* 11
+* 12
+
+
+To select a Java version, specify a `type` such as `java:11`:
+
+```yaml
+# .platform.app.yaml
+type: "java:11"
+```
+
+## Support libraries
+
+While it is possible to read the environment directly from your application, it is generally easier and more robust to use the [`platformsh/config-reader`](https://github.com/platformsh/config-reader-java) Composer library which handles decoding of service credential information for you.
+
+## Support build automation
+
+Platform.SH has support for software project management and comprehension tool in the Java World: 
+
+* [Gradle](https://gradle.org/)
+* [Maven](https://maven.apache.org/)
+* [Ant](https://ant.apache.org/)
+
+
+## Accessing services
```

Comment:
Won't be able to do pull-ins until java is added to examples.docs.platform.sh, so we'll need service access examples here, obviously.

Reply:
I'll comment that, but I believe that need to be there together with other languages.